bucket_manifest/bucket_manifest_job.py

```diff
@@ -0,0 +1,216 @@
+    """
+    Consume the sqs and write results to tsv manifest
+
+    Args:
+        queue_url(str): SQS url
+        n_total_messages(int): The expected number of messages being received
+        bucket_name(str): bucket for uploading the manifest to
+    """
+    with open("./creds.json") as creds_file:
+        creds = json.load(creds_file)
+        aws_access_key_id = creds.get("aws_access_key_id")
+        aws_secret_access_key = creds.get("aws_secret_access_key")
+
+    logging.info("Start consuming queue {}".format(queue_url))
+    # Create SQS client
+    sqs = boto3.client("sqs", region_name=REGION)
+
+    n_messages = 0
+    files = []
+    while True:
+        try:
+            # recive a message from SQS queue
+            response = sqs.receive_message(
+                QueueUrl=queue_url,
+                AttributeNames=["SentTimestamp"],
+                MaxNumberOfMessages=1,
+                VisibilityTimeout=0,
+                WaitTimeSeconds=0,
+            )
+
+            n_messages += len(response["Messages"])
+            if n_messages % 10 == 0:
+                logging.info("Received {}/{} messages".format(n_messages, n_total_messages))
+
+            for message in response["Messages"]:
+                msgBody = json.loads(message["Body"])
+                if "ERROR" in msgBody:
+                    msgBody["size"] = 0
+                    msgBody["md5"] = msgBody["ERROR"]
+                    del msgBody["ERROR"]
+                files.append(json.loads(message["Body"]))
+
+                # Delete received message from queue
+                receipt_handle = message["ReceiptHandle"]
+                sqs.delete_message(QueueUrl=queue_url, ReceiptHandle=receipt_handle)
+
+            if n_messages >= n_total_messages:
+                break
+
+        except ClientError as e:
+            logging.error(e)
+        except KeyError:
+            # Queue is empty. Check again later!!!
+            logging.info("SQS queue is empty. Taking a sleep ....")
+            time.sleep(10)
+
+    if len(files) > 0:
+        parts = urlparse(files[0]["url"])
+        now = datetime.now()
+        current_time = now.strftime("%m_%d_%y_%H:%M:%S")
+
+        filename = "manifest_{}_{}.tsv".format(parts.netloc, current_time)
+        utils.write_csv(filename, files, ["url", "size", "md5"])
+    
+        utils.upload_file(
+            filename,
+            bucket_name,
+            aws_access_key_id=aws_access_key_id,
+            aws_secret_access_key=aws_secret_access_key,
+        )
+        logging.info(
+            "Output manifest is stored at s3://{}/{}".format(bucket_name, filename)
+        )
+
+    logging.info("DONE!!!")
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser()
+    subparsers = parser.add_subparsers(title="action", dest="action")
+
+    bucket_manifest_cmd = subparsers.add_parser("create_manifest")
+    bucket_manifest_cmd.add_argument("--bucket", required=True)
```

Comment:
can  you add descriptions to each of these so --help provides some info on what to provide to the cli?

Reply:
I think the arg is `help="s3 bucket name to generate manifest for, also puts final manifest in this same bucket"` something like that